towerlib/entities/job.py

```diff
@@ -842,17 +842,34 @@ def add_credential(self, credential, credential_type='Machine'):
             credential_type (int): The type of the credential.
 
         """
-        # if not isinstance(credentials, (list, tuple)):
-        #     credentials = [credentials]
         extra_credential = self.inventory.organization.get_credential_by_name(credential,
                                                                               credential_type)
         if not extra_credential:
             raise InvalidCredential(credential)
-        payload = {'id': extra_credential.id}
+        return self._add_credential(extra_credential.id, f'Failed to add credential {credential}')
+
+    def add_credential_by_id(self, credential_id):
+        """Adds credential by name.
+
+        Args:
+            credential_id (int): The id of the credential.
+
+        """
+        return self._add_credential(credential_id, f'Failed to add credential id {credential_id}')
+
+    def _add_credential(self, credential, error_message):
```

Comment:
I think that this should be called ```_add_credential_by_id``` as that is what it actually does

Reply:
Renamed the function